package/boot/uboot-envtools/files/ramips

```diff
@@ -77,6 +77,7 @@ jcg,q20|\
 linksys,e7350|\
 netgear,eax12|\
 netgear,wax202|\
+netgear,wax214v2|\
```

Comment:
Can you show me the output of the `hexdump -C -v -n 256 /dev/mtd1`. I saw you removed the uboot-envtools package in Makefile?

Reply:
```
root@OpenWrt:/# hexdump -C -v -n 256 /dev/mtd1
00000000  97 34 66 a6 62 61 75 64  72 61 74 65 3d 31 31 35  |.4f.baudrate=115|
00000010  32 30 30 00 62 6f 6f 74  5f 6e 6d 72 70 5f 63 6c  |200.boot_nmrp_cl|
00000020  72 5f 73 74 72 5f 62 6c  6b 73 3d 6e 61 6e 64 20  |r_str_blks=nand |
00000030  65 72 61 73 65 2e 73 70  72 65 61 64 20 24 7b 73  |erase.spread ${s|
00000040  74 72 69 6e 67 74 61 62  6c 65 6f 66 66 73 65 74  |tringtableoffset|
00000050  7d 20 24 7b 73 74 72 69  6e 67 74 61 62 6c 65 66  |} ${stringtablef|
00000060  72 65 65 73 69 7a 65 7d  00 62 6f 6f 74 5f 6e 6d  |reesize}.boot_nm|
00000070  72 70 5f 77 72 5f 73 74  72 3d 6d 74 6b 75 70 67  |rp_wr_str=mtkupg|
00000080  72 61 64 65 20 73 74 72  00 62 6f 6f 74 5f 76 65  |rade str.boot_ve|
00000090  72 73 69 6f 6e 3d 32 30  32 31 30 34 30 38 00 62  |rsion=20210408.b|
000000a0  6f 6f 74 5f 77 72 5f 69  6d 67 3d 77 72 69 74 65  |oot_wr_img=write|
000000b0  69 6d 67 00 62 6f 6f 74  63 6d 64 3d 6d 74 6b 61  |img.bootcmd=mtka|
000000c0  75 74 6f 62 6f 6f 74 00  62 6f 6f 74 64 65 6c 61  |utoboot.bootdela|
000000d0  79 3d 30 00 64 65 63 72  79 70 74 5f 72 65 73 75  |y=0.decrypt_resu|
000000e0  6c 74 3d 67 6f 6f 64 00  65 74 68 61 63 74 3d 65  |lt=good.ethact=e|
000000f0  74 68 40 31 65 31 30 30  30 30 30 00 65 74 68 61  |th@1e100000.etha|
00000100
root@OpenWrt:/# 
```

This is the `fw_printenv` output:

```
root@OpenWrt:/# fw_printenv
baudrate=115200
boot_nmrp_clr_str_blks=nand erase.spread ${stringtableoffset} ${stringtablefreesize}
boot_nmrp_wr_str=mtkupgrade str
boot_version=20210408
boot_wr_img=writeimg
bootcmd=mtkautoboot
bootdelay=0
decrypt_result=good
ethact=eth@1e100000
ethaddr=94:18:65:18:E7:6F
fdtcontroladdr=9bfef730
fenv_factory=off
fenv_model=WAX214v2
fenv_region=NA
fenv_restore=yes
fileaddr=80010000
filesize=8a0000
filesize_result=good
invaild_env=no
ipaddr=192.168.1.1
mtddevname=u-boot
mtddevnum=0
mtdids=nand0=nand0
mtdparts=mtdparts=nand0:512k(u-boot),512k(u-boot-env),512k(factory),38912K(firmware),38912K(firmware_backup),13312K(CRP),4096K(Language)
netmask=255.255.255.0
partition=nand0,0
serverip=192.168.1.2
stderr=uartlite0@1e000c00
stdin=uartlite0@1e000c00
stdout=uartlite0@1e000c00
root@OpenWrt:/# 
```

The same `fw_printenv` output is seen from the stock firmware.

Without this change, this is the output:

```
root@OpenWrt:/# fw_printenv
Cannot parse config file '/etc/fw_env.config': No such file or directory
Failed to find NVMEM device
root@OpenWrt:/# 
```

#10366 has information about why this change was needed for the Netgear WAX202. The contributor of that change suggested that it is probably safer to leave the Config partition read-only, and I figured that they removed the uboot-envtools package for the WAX202 since changes would not be able to be made. I followed suit when adding support for the WAX214v2; would a different approach be better?

Reply:
Hi! Please use u-boot-env `ethaddr` MAC address as the base MAC instead of the `mac`. In this way, we can move MAC settings to dts in the future.

Reply:
Okay, done

Reply:
Do you want to switch to `NVMEM_U_BOOT_ENV` already now, or should we do it later for all 3 devices?

Reply:
I don't mind either way (but I would need help if we should do it for this device now), but I found https://github.com/openwrt/openwrt/pull/14182 - should we wait for this to be finished first?

Reply:
+1 for u-boot,env. That PR is stalled. Might as well do it here.

Reply:
Okay, I've made an attempt at switching to `NVMEM_U_BOOT_ENV` and `compatible = "u-boot,env";` in the latest push, and have attached the latest OpenWrt boot log in the PR description (for commit 8a515d4300e423e00d7e7fc0b643bfe17e349f5f).

One problem I encountered is that removing [this line](https://github.com/openwrt/openwrt/commit/8a515d4300e423e00d7e7fc0b643bfe17e349f5f#diff-a4a13db5c96026e0a9d0909fa19257b892affe1f5944275a7f4ce851eb90ca79R82) made `fw_setenv` fail with the following error:

```
root@OpenWrt:/# fw_setenv bootdelay 5
Can't open /sys/bus/nvmem/devices/u-boot-env0/nvmem: Permission denied
Error: can't write fw_env to flash
```

Leaving this line in made `fw_setenv` work properly again (a test of this is in the latest OpenWrt boot log). I'm not sure if there is something I was missing.

Reply:
So mac address is correct without the line?

Reply:
I rebuilt without the line again as a test and the mac address is correct without it.

Reply:
Weird. Does fw_printenv work without it too?

Reply:
Yes, fw_printenv works with or without the line.

Reply:
I would guess envtools needs a patch to deal with this then. ping @Ansuel @rmilecki

Reply:
Sorry to do this, but please revert. I've confirmed the nvmem uboot env driver does not support writing. This will have to wait for some other time.

In other news, I restarted the other PR.

Reply:
Okay, I've reverted the nvmem changes. A new OpenWrt boot log has been added to the description (for commit f212d365bae7fbd75913b65f31296480d86171c0).